### ISSUE TITLE
pdf syncing broken

### ISSUE THREAD

**Issue by aaltland** [SublimeText/LaTeXTools#919]
As of the latest revision of latextools (v3.12.5) pdf-syncing no longer works. I am on a mac using skim as a previewer. The sublime text/pdf sync is eternally frozen at the latest position in my pdf doc where syncing still worked before last night's latextools update.

**Comment by r-stein** [comment]
- Have you also updatet skim, since there seem to be an issue with that https://github.com/SublimeText/LaTeXTools/issues/876
- Try to run `C-l, backspace` to remove all sync files

**Comment by aaltland** [comment]
I updated skim this morning. C-l backspace doesn’t fix it :(

**Comment by aaltland** [comment]
Following the advise given above I installed the older version of skim, 1.4.21. Problem remains

**Comment by aaltland** [comment]
I just checked on another mac on which sublime/latextools/skim are running. Same phenomenon, as of this morning.

**Comment by r-stein** [comment]
You could check the ST console ``ctrl+` `` for errors while pressing forward sync.

**Comment by aaltland** [comment]
It shows:
Jump to:  5101 10
<class '_latextools_skim_viewer.SkimViewer'>
Running "/Applications/Skim.app/Contents/SharedSupport/displayline -r 5102 /Users/alexaltland/Dropbox/revtex/LEHRE/TP0/TEXT/cha1.pdf /Users/alexaltland/Dropbox/revtex/LEHRE/TP0/TEXT/cha1.tex"

The line number(5101) is the one from which I triggered the command. However, I end up at the same frozen position in my pdf as always.

**Comment by aaltland** [comment]
If I run the same command from terminal I get the same result.

**Comment by aaltland** [comment]
I just checked with another document. For that one syncing worked! The difference is that the one for which it was broken is part of a multi-file doc managed by the latex package subfiles. Compiling a single chapter with the latextools multifile support (%!TEX root=main.tex) **switched off** produces the error. I activated that line and now syncing works! Comments: 1) strange that it used to work until yesterday for a single subfiles doc 2) it sucks to run a 800 pages doc with  %!TEX root=main.tex switched on, compiling is terribly slow :(. Anyway, the problem seems to be solved, thanks for the support!

**Comment by aaltland** [comment]
The problem with the syncing I ran into this morning remains. I now understand the situation better.

— For single file documents pdf-sync works fine.

— The problem is with multi file documents managed by the subfiles package. 

— I have a document main.tex which has cha1.tex as a subfile. When I build cha1.tex with latextools the following build command ensues:

['latexmk', '-cd', '-f', '-pdf', '-interaction=nonstopmode', '-synctex=1', 'main.tex']
Running “latexmk -cd -f -pdf -interaction=nonstopmode -synctex=1 main.tex"

— This is not the command I want. I should be latexmk on cha1.tex. Until yesterday, it was like this, and I got cha1.pdf. However, now main.pdf is produced

— The viewer command then triggered by latextools is

Running "/Applications/Skim.app/Contents/SharedSupport/displayline -r 1 /Users/alexaltland/Dropbox/revtex/LEHRE/TP0/TEXT/main.pdf /Users/alexaltland/Dropbox/revtex/LEHRE/TP0/TEXT/cha1.tex"


This is where skim gets confused. It cannot handle the main.pdf in combination with cha1.tex as a text file.

For the time being, I have deactivated %!TEX root=main.tex, however the problem remains with that line included.

Any comments? I hesitate opening another issue, don’t want to create confusion.

Best, A.

**Comment by aaltland** [comment]
The command calling skim in the multifile doc reads:

Running “/Applications/Skim.app/Contents/SharedSupport/displayline -r 5332 /Users/alexaltland/Dropbox/revtex/LEHRE/TP0/TEXT/cha1.pdf /Users/alexaltland/Dropbox/revtex/LEHRE/TP0/TEXT/cha1.tex"

This does not work.

The command in the single file doc for which syncing works:

Running “/Applications/Skim.app/Contents/SharedSupport/displayline -r 1529 '/Users/alexaltland/Dropbox/revtex/MANY BODY LOCALIZATION/note.pdf' '/Users/alexaltland/Dropbox/revtex/MANY BODY LOCALIZATION/note.tex'"

Notice the quotes around the file names which are missing in the broken command. Could this cause the problem?

A.

**Comment by ig0774** [comment]
How, exactly, are you "deactivating" `%!TEX root=main.tex`? I ask because essentially only two things will disable that line: 1) adding a space between the `!` and the `TEX` or 2) deleting the line altogether. (This hasn't changed in quite a few versions).

We have no special support for subfiles per se, but things *should* work as long as LaTeXTools doesn't think you've set a TeX root when trying to compile a sub-document. From the sub-document, you can use the **LaTeXTools: System Check** command to see what LaTeXTools thinks the root document is. If you want to compile, e.g., `cha1`, it should be `cha1.tex`, etc.

**Comment by aaltland** [comment]
got it, I thought I could deactivate by adding another %. Now removed the line and cha1.tex compiles into cha1.pdf. Great. However, sync still doesn’t work, and I have a guess why: see the view command below:

Running “/Applications/Skim.app/Contents/SharedSupport/displayline -r 5803 /Users/alexaltland/Dropbox/revtex/LEHRE/TP0/TEXT/cha1.pdf /Users/alexaltland/Dropbox/revtex/LEHRE/TP0/TEXT/cha1.tex"

sync broken.

Now compare to the sync command for a SINGLE FILE doc, for which sync works fine:

Running “/Applications/Skim.app/Contents/SharedSupport/displayline -r 1660 '/Users/alexaltland/Dropbox/revtex/MANY BODY LOCALIZATION/note.pdf' '/Users/alexaltland/Dropbox/revtex/MANY BODY LOCALIZATION/note.tex'"

Notice the quotes around the file names, present in the working command, absent in the broken one. The only difference between the compiles is that the first is on a subfile, the other one on a single file. 

Could this be the cause of the problem?

A.

**Comment by aaltland** [comment]
guys, the problem is fixed. It had nothing to do with the quotes. The document I am working on is under subversion control. It looks that my collaborator had placed an obscure macro command somewhere in the middle which, for reasons I don’t understand, caused pdf-syncing to stop working. This happened yesterday night in parallel with the update. I am awfully sorry for the trouble-making! Embarrassing, two things happening at once I was too dumb to spot it. Apologies, A.

**Comment by r-stein** [comment]
ok, nice it works now =)
